apps/vscode/vscode.py

```diff
@@ -170,53 +170,183 @@ def command_palette():
         actions.key("cmd-shift-p")
 
 
+def _replace_tab_with_untitled_file():
+    """Simulates the creation of an empty split by replacing a single tab in a split group with an untitled file"""
+    actions.user.vscode("workbench.action.files.newUntitledFile")
+    actions.app.tab_previous()
+    actions.user.tab_close_wrapper()
+
+
 @ctx.action_class("user")
-class UserActions:
-    # splits.py support begin
-    def split_clear_all():
-        actions.user.vscode("workbench.action.editorLayoutSingle")
+class SplitActions:
+    # Creation
 
-    def split_clear():
-        actions.user.vscode("workbench.action.joinTwoGroups")
+    ## Empty splits
+    def split_create():
+        actions.user.split_clone()
+        _replace_tab_with_untitled_file()
 
-    def split_flip():
-        actions.user.vscode("workbench.action.toggleEditorGroupLayout")
+    def split_create_right():
+        actions.user.split_clone_right()
+        _replace_tab_with_untitled_file()
 
-    def split_maximize():
-        actions.user.vscode("workbench.action.maximizeEditor")
+    def split_create_left():
+        actions.user.split_clone_left()
+        _replace_tab_with_untitled_file()
 
-    def split_reset():
-        actions.user.vscode("workbench.action.evenEditorWidths")
+    def split_create_down():
+        actions.user.split_clone_down()
+        _replace_tab_with_untitled_file()
 
-    def split_last():
+    def split_create_up():
+        actions.user.split_clone_up()
+        _replace_tab_with_untitled_file()
+
+    ## Duplicated splits
+    def split_clone():
+        actions.user.vscode("workbench.action.splitEditor")
+
+    def split_clone_right():
+        actions.user.vscode("workbench.action.splitEditorRight")
+
+    def split_clone_left():
+        actions.user.vscode("workbench.action.splitEditorLeft")
+
+    def split_clone_down():
+        actions.user.vscode("workbench.action.splitEditorDown")
+
+    def split_clone_up():
+        actions.user.vscode("workbench.action.splitEditorUp")
+
+    def split_clone_vertically():
+        actions.user.vscode("workbench.action.splitEditor")
+
+    def split_clone_horizontally():
+        actions.user.vscode("workbench.action.splitEditorOrthogonal")
+
+    # Destruction
+    def split_close():
+        actions.user.vscode("workbench.action.joinTwoGroups")
+
+    def split_close_all():
+        actions.user.vscode("workbench.action.editorLayoutSingle")
+
+    # Navigation
+    def split_focus_previous():
         actions.user.vscode("workbench.action.focusLeftGroup")
```

Comment:
workbench.action.focusLastEditorGroup

Reply:
Should actually be workbench.action.focusPreviousGroup